python/python/glide/async_commands/server_modules/ft_options/ft_create_options.py

```diff
@@ -0,0 +1,429 @@
+    """
+    Abstract base class for defining vector field attributes to be used after the vector algorithm name.
+    """
+
+    @abstractmethod
+    def __init__(self, dim: int, distance_metric: DistanceMetricType, type: VectorType):
+        """
+        Initialize a new vector field attributes instance.
+
+        Args:
+            dim (int): Number of dimensions in the vector.
+            distance_metric (DistanceMetricType): The distance metric used in vector type field. Can be one of [L2 | IP | COSINE].
+            type (VectorType): Vector type. The only supported type is FLOAT32.
+        """
+        self.dim = dim
+        self.distance_metric = distance_metric
+        self.type = type
+
+    @abstractmethod
+    def toArgs(self) -> List[str]:
+        """
+        Get the arguments to be used for the algorithm of the vector field.
+
+        Returns:
+            List[str]: A list of arguments.
+        """
+        args = []
+        if self.dim:
+            args.extend([FtCreateKeywords.DIM, str(self.dim)])
+        if self.distance_metric:
+            args.extend([FtCreateKeywords.DISTANCE_METRIC, self.distance_metric.name])
+        if self.type:
+            args.extend([FtCreateKeywords.TYPE, self.type.name])
+        return args
+
+
+class VectorFieldAttributesFlat(VectorFieldAttributes):
+    """
+    Get the arguments to be used for the FLAT algorithm of the vector field.
+    """
+
+    def __init__(
+        self,
+        dim: int,
+        distance_metric: DistanceMetricType,
+        type: VectorType,
+        initial_cap: Optional[int] = None,
+    ):
+        """
+        Initialize a new flat vector field attributes instance.
+
+        Args:
+            dim (int): Number of dimensions in the vector.
+            distance_metric (DistanceMetricType): The distance metric used in vector type field. Can be one of [L2 | IP | COSINE].
+            type (VectorType): Vector type. The only supported type is FLOAT32.
+            initial_cap (Optional[int]): Initial vector capacity in the index affecting memory allocation size of the index. Defaults to 1024.
+        """
+        super().__init__(dim, distance_metric, type)
+        self.initial_cap = initial_cap
+
+    def toArgs(self) -> List[str]:
+        """
+        Get the arguments representing the vector field created with FLAT algorithm.
+
+        Returns:
+            List[str]: A list of FLAT algorithm type vector arguments.
+        """
+        args = super().toArgs()
+        if self.initial_cap:
+            args.extend([FtCreateKeywords.INITIAL_CAP, str(self.initial_cap)])
+        return args
+
+
+class VectorFieldAttributesHnsw(VectorFieldAttributes):
+    """
+    Get the arguments to be used for the HNSW algorithm of the vector field.
+    """
+
+    def __init__(
+        self,
+        dim: int,
+        distance_metric: DistanceMetricType,
+        type: VectorType,
+        initial_cap: Optional[int] = None,
+        m: Optional[int] = None,
```

Comment:
I dont like the naming, should be something more informitive

Reply:
I see this is the naming of memdb as well, I dont understand why such bad naming 😞

Reply:
I considered naming based on the what the field does but decided to not diverge from the documentation.